src/QAT_RPC/qat_rpc/zmq/wrappers.py

```diff
@@ -65,17 +75,56 @@ def __init__(self, hardware: QuantumHardwareModel = None):
     @property
     def address(self):
         return f"{self._protocol}://*:{self._port}"
+    
+    def _program(self, program, config):
+        program = program
+        config = CompilerConfig.create_from_json(config)
+        result, metrics = execute_with_metrics(program, self._engine, config)
+        return {"results": result, "execution_metrics": metrics}
+
+    def _version(self):
+        return {"qat_rpc_version": str(version('qat_rpc'))}
+    
+    def _couplings(self):
+        coupling =  [coupled.direction for coupled in self._hardware.qubit_direction_couplings]
+        return {"couplings": coupling}
+
+    def _qubit_info(self):
+        raise NotImplementedError("Individual qubit information not implented, pending hardware model changes.")
+
+    def _qpu_info(self):
+        features = OpenPulseFeatures()
+        features.for_hardware(self._hardware)
+        qpu_info = features.to_json_dict()
+        return {"qpu_info": qpu_info}
+    
+    def _interpret_message(self, message):
+        match message[0]:
```

Comment:
This syntax is not supported in python < 3.10

Reply:
Looks like we are accepting qat-rpc is now python 3.10+ ?